### PR TITLE
Added SimpleBinary binding

### PR DESCRIPTION
**Pull request by docbender**
New merged binding pull request ready to review. Original pull request was #3589.

### CODE REVIEW

File: bundles/binding/org.openhab.binding.simplebinary/src/main/java/org/openhab/binding/simplebinary/internal/SimpleBinaryUART.java [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+    /**
+     * Write data into device stream
+     *
+     * @param data
+     *            Item data with compiled packet
+     */
+    @Override
+    protected boolean sendDataOut(SimpleBinaryItemData data) {
+        // data line stabilization
+        if (LINE_STABILIZATION_TIME > 0) {
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.simplebinary/src/main/java/org/openhab/binding/simplebinary/internal/SimpleBinaryUART.java]: This block can never be executed. (The "variable" is always 0)

File: bundles/binding/org.openhab.binding.simplebinary/src/main/java/org/openhab/binding/simplebinary/internal/SimpleBinaryProtocol.java [hunk 2]
```diff
@@ -0,0 +1,799 @@
...
+            Map<String, SimpleBinaryBindingConfig> itemsConfig, String deviceName, boolean letDataInBuffer)
+            throws NoValidCRCException, NoValidItemInConfig, UnknownMessageException, ModeChangeException {
+        byte devId = data.get();
+        byte msgId = data.get();
+        int address = 0;
+        byte[] rawPacket;
+        byte[] rawData = null;
+        byte crc = 0, calcCrc;
+
+        try {
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.simplebinary/src/main/java/org/openhab/binding/simplebinary/internal/SimpleBinaryProtocol.java]: This try block catches nothing. Why is it here?

File: bundles/binding/org.openhab.binding.simplebinary/src/main/java/org/openhab/binding/simplebinary/internal/SimpleBinaryUART.java [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+                timeoutTask.cancel();
+            }
+
+            timeoutTask = new TimerTask() {
+                @Override
+                public void run() {
+                    timeoutTask = null;
+                    dataTimeouted();
+                    setWaitingForAnswer(false);
+                    // TODO: should i call it????
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.simplebinary/src/main/java/org/openhab/binding/simplebinary/internal/SimpleBinaryUART.java]: Remove or clarify

File: bundles/binding/org.openhab.binding.simplebinary/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ org.openhab.core.types,
+ org.openhab.model.item.binding,
+ org.osgi.framework,
+ org.osgi.service.component,
+ org.osgi.service.event,
+ org.slf4j
+Export-Package: org.openhab.binding.simplebinary
+Bundle-DocURL: http://www.openhab.org
+Bundle-RequiredExecutionEnvironment: J2SE-1.5
+Service-Component: OSGI-INF/binding.xml, OSGI-INF/genericbindingprovider.xml
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.simplebinary/META-INF/MANIFEST.MF]: Bundle-ActivationPolicy should be set to lazy

**Review by 9037568 — CHANGES_REQUESTED** [review]
Thanks for this contribution.

In addition to my inline comments:

Please run all the source through the Eclipse formatter.

Also, in a number of places you are using this construct:
````
       if (logger.isDebugEnabled()) {
            logger.debug("execute() method is called!");
       }
````

This is really not worth doing. The wrapper that checks the logging level should only be used around operations that are expensive. You should consider eliminating these if conditionals anywhere that they're only wrapping a call to logger.x().

**Comment by docbender** [comment]
Hello, 
thanks for your review.

About _logger_ section. I had a problem in production environment (logger level=WARN) with heavy traffic that sometimes totally random it just crash. I found that it was at _logger.debug(...)_ line.  With condition _logger.isDebugEnabled()_ this crashes disappeared.

Another one. Why should I change year in header? Some files was created in 2015. Same with openhab version. Pull request of this binding was pulled a year ago before 1.8.0 was released.

Finally complete all missing header doc would be long time run.

**Comment by 9037568** [comment]
> About logger section.

That's merely advisory. If you don't want to change those, it'll still be fine to merge.

> Another one. Why should I change year in header?

As far as I'm aware, the convention is to have the dates in the copyright headers be up to date based on the year the code is first merged. Certainly the openhab version in the "since" tags needs to be correct. It's also a quick and easy fix.

> Finally complete all missing header doc would be long time run.

If this is about the "@return" doc comments that are incomplete/empty, again I'd call those advisory. I think they'd still be ok to merge if you choose not to clean them up.

**Comment by docbender** [comment]
Modified code pushed in. Thanks for review

**Comment by 9037568** [comment]
Thanks for these updates. There are still a few issues of commented code and copyright dates that need to be addressed, however.

**Comment by 9037568** [comment]
I just realized while perusing the wiki page for this binding that you need to add your configuration items to the [openhab_default.cfg](https://github.com/openhab/openhab/blob/master/distribution/openhabhome/configurations/openhab_default.cfg) file.

**Comment by docbender** [comment]
I pushed openhab_default.cfg update. I also corrected logger strings and the wrong copyright date in OSGI files.

**Comment by 9037568** [comment]
Good progress. Still a few left.

**Pull request closed without merging**